🐛 BUG: Large File Uploads with Local Mode

### Which Cloudflare product(s) does this pertain to?

R2, Wrangler core

### What version(s) of the tool(s) are you using?

3.55.0[Wrangler]

### What version of Node are you using?

22.1.0

### What operating system and version are you using?

MacOS Sonoma 14.4.1

### Describe the Bug

### Observed behavior
When attempting to "upload" a large object to a local R2 instance via `wrangler r2 object put <obj_name>`, it can fail with the error `✘ [ERROR] Error: Wrangler only supports uploading files up to 300 MiB in size`. While this makes sense when pushing to a remote bucket due to API size limits, it doesn't make as much sense when in local mode.

### Expected behavior
`wrangler` should be able to push objects above the max size limits when in local mode.

### Steps to reproduce
1. Create `wrangler` project with R2 Binding.
2. Create/find file above 300 MB limit.
3. Run `wrangler r2 object put BUCKET_NAME/directory/file.name -f=YOUR_FILE.name --local`

I have a one-line fix on [my fork](https://github.com/helloimalastair/workers-sdk/tree/r2-local), but remembered to make an issue before submitting a PR.

--- a/.changeset/wise-ravens-mate.md
+++ b/.changeset/wise-ravens-mate.md
@@ -1,0 +1,5 @@
+---
+"wrangler": patch
+---
+
+Wrangler is now able to upload files to local R2 buckets above the 300 MiB limit

--- a/packages/wrangler/src/r2/index.ts
+++ b/packages/wrangler/src/r2/index.ts
@@ -279,7 +279,7 @@ export function r2(r2Yargs: CommonYargsArgv) {
 							objectSize = blob.size;
 						}
 
-						if (objectSize > MAX_UPLOAD_SIZE) {
+						if (objectSize > MAX_UPLOAD_SIZE && !local) {
 							throw new FatalError(
 								`Error: Wrangler only supports uploading files up to ${prettyBytes(
 									MAX_UPLOAD_SIZE,

